helper/chksum.c

```diff
@@ -219,11 +221,15 @@ static inline int odph_process_l3_hdr(odp_packet_t odp_pkt,
 			ipv6_hdr_ptr = &ipv6_hdr;
 		}
 
-		addrs_ptr        = (uint16_t *)(void *)&ipv6_hdr_ptr->src_addr;
-		addrs_len        = 2 * ODPH_IPV6ADDR_LEN;
 		protocol         = ipv6_hdr_ptr->next_hdr;
 		ipv6_payload_len = odp_be_to_cpu_16(ipv6_hdr_ptr->payload_len);
 		l3_len           = ipv6_payload_len + ODPH_IPV6HDR_LEN;
+		for (idx = 0; idx < 8; idx++) {
```

Comment:
The indexing is wrong. Only the first 9 bytes of the addresses are looked at and most of them twice.

This would not look right to me even with fixed indexing. This code produces the same result regardless of the endianess of the CPU whereas the old code produces different results in big and little endian CPUs.

Reply:
Yes.. Will be fixed in the new patch-set